backend/notifications/notificationGenerator.js

```diff
@@ -0,0 +1,155 @@
+import { prisma } from "../utils/helpers.js";
+
+const generateWorkoutReminder = async (userId) => {
+  const user = await prisma.user.findUnique({
+    where: { id: userId },
+    include: { userInterests: true },
+  });
+
+  if (!user || !user.userInterests) return null;
+
+  const daysSinceLastWorkout = Math.floor(
+    (Date.now() - user.userInterests.lastWorkoutDate.getTime()) /
+      (1000 * 3600 * 24)
+  );
+
+  if (daysSinceLastWorkout > 3) {
+    return {
+      type: "workout_reminder",
+      content: `It's been ${daysSinceLastWorkout} days since your last workout. Time to get moving!`,
+    };
+  }
+  return null;
+};
+
+const generateTemplateRecommendation = async (userId) => {
+  const userInterest = await prisma.userInterest.findUnique({
+    where: { userId },
+    include: { interests: true },
+  });
+
+  if (!userInterest || userInterest.interests.length === 0) return null;
+
+  const highestInterest = userInterest.interests.reduce((max, interest) =>
+    interest.score > max.score ? interest : max
+  );
+
+  if (highestInterest.score > 70) {
+    const recommendedTemplate = await prisma.workoutTemplate.findFirst({
```

Comment:
This is not a blocker but for better user experience: to make sure that the notif is not too spammy for users, we should also try to filter out templates that we've already recommended before.